develop-docs/sdk/telemetry/traces/tracing-without-performance.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Tracing Without Performance
+---
+
+If Sentry SDKs are not configured for sending spans (or transactions), they should fall back to a mode where they still handle attaching and propagating trace data.
+Internally, we call this mode "Tracing without Performance" (or "TwP" in short).
+TwP mode ensures that error events and other signals are still connected via a trace, even if users choose not to send spans or performance data.
+
+## Outcome
+
+Tracing without Performance is the default fallback behavior of SDKs, when users [don't specify any tracing options](#configuration).
+
+This means that SDKs in this mode always:
+
+- continue incoming traces
+- attache `event.contexts.trace` context on events (e.g. errors, check-ins)
+- attach the `trace` envelope header to Sentry envelopes, populated from the dynamic sampling context
+- propagate trace data (`sentry-trace`, `baggage`) via the usual channels (e.g. HTTP headers, `<meta> HTML tags`, etc), with the correct [sampling decision](#sampling-decision)
+
+If users [explicitly opt out](#completely-disable-tracing) of all tracing capabilities, TwP mode will no longer be active.
+
+## Configuration
+
+Twp is active by default if users don't specify sampling options that enable span collection and sending.
+For example, the following trivial SDK setup (note the absence of sampling options) enables TwP:
+
+```JavaScript
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+});
+```
+
+TwP mode can be disabled by either "upgrading" to Tracing _with_ Performance (see below) or by entirely opting out of trace propagation.
+
+<Alert>
+
+Note that some SDKs like the JavaScript Browser SDK require additional integrations to enable tracing all together.
+This is fine as long as there's a good reason why additional configuration is required (like keeping bundle size in check in JS).
+
+</Alert>
+
+### Upgrading to Tracing With Performance
+
+As soon as users specify a tracing option in the SDK configuration, SDKs will use the normal tracing mode, which means collecting and sending spans (or transactions).
+
+This can be achieved by specifying `tracesSampleRate` or `tracesSampler`, or any other equivalent option in the SDK configuration:
+
+```js
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+  tracesSampleRate: 0.1,
+  // or
+  tracesSampler: () => {
+    /*...*/
+  },
+});
+```
+
+It's important to note that setting `tracesSampleRate: 0` disables TwP mode.
+
+### Completely Disabling Tracing
+
+To completely opt out of any distributed tracing capabilities, users can pass an empty array (or language-equivalent parameter) to the `tracePropagationTargets` option. This will prevent the SDK from propagating trace information any further.
+
+```js
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+  tracePropagationTargets: [],
+});
+```
```

Comment:
Might want to call out `tracesSampleRate: null`, too?

Reply:
What should `tracesSampleRate: null` do? At least in JS this isn't a valid value. If we set `tracesSampleRate: undefined` TwP is still active.

Reply:
So `null/nil/undefined` all has the same behaviour. It does no longer take trace parent headers with `-1` into account at all.

Reply:
Sorry but I'm a bit confused:

This section is about disabling tracing without performance, i.e. no longer propagating a trace. As far as I can tell, this is only the case when setting `tracePropagationTargets: []`. 

> It does no longer take trace parent headers with -1 into account at all

What does this mean specifically? If an incoming SDK receives a header with `-1`, and its `tracesSampleRate` is `null|undefined|nil`, should the SDK _not_ continue the positive sampling decision? Maybe I'm missing something obvious but I currently don't see this behaviour in the SDK code. 

To further add to the confusion, we would like to ensure that in the future, setting `tracesSampleRate: undefined` behaves in the same way as not setting `tracesSampleRate` at all. It's a bit intransparent to users to understand that there is difference while in both cases querying `options.tracesSamplerate` would return `undefined`.

Reply:
If this entire page only focuses on TwP, then the section header (Completely Disabling Tracing) confuses me 😬 

> What does this mean specifically? If an incoming SDK receives a header with -1, and its tracesSampleRate is null|undefined|nil, should the SDK not continue the positive sampling decision?

Yes, this is how it should work today.

We'll properly fix this when we re-work all SDKs towards only spans and the new sampling. `tracesSampleRate` will only be a float between `0.0` and `1.0`.

Reply:
> then the section header (Completely Disabling Tracing) confuses me

Fair but I think it's important to specify how you opt out of TwP. There are two options:
1. Enable regular tracing (with spans) - i.e. specify `tracesSampleRate: number` or `tracesSampler`. 
2. Completely disabling tracing. Should we rename the section? I was also thinking about "Disabling Tracing without Performance" but this could also entail 1... Open for better suggestions! 

> Yes, this is how it should work today.

Okay, so I think we're talking about 2 things:

- Assuming no other tracing options (`tracesSampler`/`enableTracing`) are set: TwP should be enabled if `tracesSampleRate` is not set at all or if `tracesSampleRate` is set to `undefined/null/nil`. 
I think that makes sense to me and is what we want to achieve in v9 of the JS SDK anyway. Today, there's a behavioural difference between setting `tracesSampleRate: undefined` and not setting it at all. But this shouldn't affect the develop spec. 

- Separately to this, we should document that SDKs in TwP mode, that receive incoming traces with `-1` (and `-0` as well?), should **not** forward the sampling decision, meaning, propagate `sentry-trace` with a deferred sampling decision. This should go under another section though because it isn't connected to Disabling TwP.

Reply:
As discussed in the call yesterday:

- Removed the "completely disable tracing" section in favour of "disable trace propagation"
- Noted explicitly that at this time, SDKs will always continue incoming traces in TwP mode. The only way to opt out of this would be to manually start a new trace (but this is admittedly a niche edge case, given that users would probably only do this if they'd also like to send spans). 
- Realized I already added a "historical context" section at the bottom but I now added a link on the top to it. Slightly reworked the historical context section though.